nova_vm/src/heap/heap_gc.rs

```diff
@@ -165,220 +120,244 @@ pub fn heap_gc(heap: &mut Heap) {
         strings.sort();
         strings.iter().for_each(|&idx| {
             let index = idx.into_index();
-            if let Some(marked) = bits.strings.get(index) {
-                if marked.load(Ordering::Acquire) {
+            if let Some(marked) = bits.strings.get_mut(index) {
+                if *marked {
                     // Already marked, ignore
                     return;
                 }
-                marked.store(true, Ordering::Relaxed);
+                *marked = true;
             }
         });
         let mut symbols: Box<[SymbolIndex]> = queues.symbols.drain(..).collect();
         symbols.sort();
         symbols.iter().for_each(|&idx| {
             let index = idx.into_index();
-            if let Some(marked) = bits.symbols.get(index) {
-                if marked.load(Ordering::Acquire) {
+            if let Some(marked) = bits.symbols.get_mut(index) {
+                if *marked {
                     // Already marked, ignore
                     return;
                 }
-                marked.store(true, Ordering::Relaxed);
+                *marked = true;
                 let data = heap.symbols.get(index).unwrap().as_ref().unwrap();
                 if let Some(string_index) = data.descriptor {
                     queues.push_value(string_index.into());
                 }
             }
         });
-        let mut e_2_4: Box<[ElementIndex]> = queues.e_2_4.drain(..).collect();
+        let mut e_2_4: Box<[(ElementIndex, u32)]> = queues.e_2_4.drain(..).collect();
```

Comment:
Macros over all of these element queues could make this code less ugly, but it isn't strictly necessary.

Reply:
Yeah, I'm unfortunately really bad at macros so I don't really use them :D